Automattic/jetpack
# [Try] Admin Page: switch to WordPress' own toggle instead of ours

#### Changes proposed in this Pull Request:

Experiment with switching from our own Form toggle to the one available in `@wordpress/components`. Since our implementation has additional properties, let's update our implementation to keep those additional options while wrapping the `@wordpress/components` components.

Related: https://github.com/Automattic/wp-calypso/pull/46517/

- [ ] There is no more toggling state, that is something we should reintroduce.
- [ ] Does it work well [when a module is forced to be off](https://jetpack.com/support/module-overrides/)?
- [ ] Does it look good on mobile?
- [ ] Is the component documentation up to date?
- [ ] Are the margins around the toggles correct?

<img width="697" alt="screenshot 2020-10-26 at 17 08 01" src="https://user-images.githubusercontent.com/426388/97201527-d9c33a80-17b2-11eb-8f49-1796a7d3c9cb.png">
<img width="821" alt="screenshot 2020-10-26 at 17 07 41" src="https://user-images.githubusercontent.com/426388/97201533-dc259480-17b2-11eb-8887-109f7c128a06.png">
<img width="1142" alt="screenshot 2020-10-26 at 14 54 47" src="https://user-images.githubusercontent.com/426388/97201539-dcbe2b00-17b2-11eb-9a0f-9ac486204206.png">


#### Jetpack product discussion

* p4TIVU-9wx-p2

#### Does this pull request change what data or activity we track or use?

* No 

#### Testing instructions:

* Go to Jetpack > Dashboard and Jetpack > Settings
* Play with the different toggles
* Try [restricting access to a module](https://jetpack.com/support/module-overrides/), and see how the toggle behaves

#### Proposed changelog entry for your changes:

* Dashboard: use WordPress components to display toggles
